File: CursoTemplate/ClientApp/src/app/data/model/catalogo/rol-pagina.ts
import { ThemePalette } from "@angular/material/core";

export interface RolPagina {
  rolNombre: string;
  rolId: number;
  paginaNombre: string;
  paginaId: number;
  activo: boolean;
}

export interface RolPaginaListCheck {
  name: string;
  completed: boolean;
  color: ThemePalette;
  listaPagina?: RolPagina[];
}

File: ContextoDB/Models/Catalogos/RolPagina.cs

namespace ContextoDB.Models.Catalogos
{
    using ContextoDB.Models.Seguridad;

    public class RolPagina
    {
        public int RolId { get; set; }
        public int PaginaId { get; set; }
        public bool Activo { get; set; }

        public Rol Rol { get; set; }
        public Pagina Pagina { get; set; }

    }
}
